### PR TITLE
[478] - Confirm new ECT details

### LINKED ISSUE #478: Confirm new ECT details
## Description
This story covers the last step in the 'Add ECT' journey of submitting the details for the new ECT, and displaying this new ECT in the list of ECTs added to a school.

## User story

**As** a school user
**I need** to submit the ECT details I've entered
**So that** the teacher can be registered as a new ECT at my school

## Prototype, screenshots & content
[Link to prototype](https://ecf2-prototy-v3-oz6554dztlkwtv.herokuapp.com/v3/school/home/ects)
Password: review
Note, you need to select a mentor in the journey to get to the success screen.

Screenshots of relevant pages:

Image

Content should be as per the screenshots above.
URL and validation messages should be as per [this doc](https://educationgovuk.sharepoint.com/:x:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/Register%20early%20career%20teachers/Beta/Design/Register%20ECTs%20page%20titles%20and%20URLs.xlsx?d=w0c1578ea643147b29357c7f610ee178f&csf=1&web=1&e=U8iIdi).

## Acceptance criteria

- [ ] 1. When user clicks 'Confirm details' on the 'Check your answers before submitting' page, ECT details will be saved and user will be navigated to the Confirmation page.
- [ ] 2. On the Confirmation page:
- [ ] 2a. The ECT's name will be pulled into the confirmation message e.g. You have registered [John Smith] as an early career teacher
- [ ] 2b. There will be a link to go 'Back to your ECTs', which navigates the user to the 'View list of ECTs at your school' page (#468)
- [ ] 2c. There will be no 'Back' link on the Confirmation page.
- [ ] ~~3. When an ECT has been added and user goes to the 'View list of ECTs at your school' page, the ECT will be shown in the list of ECTs at that school, including their name, TRN, date of birth and email.~~
- [ ] 4. Session data should be destroyed at the start of the 'add ECT' journey

## Technical notes
ECTs will be saved in session data - writing to the database is **not** in scope for this ticket.

## Linked tickets
#468 
#475

### PR DESCRIPTION
**Pull request by TobyRet**
This is last step in the 'Add ECT' journey of submitting the details for the new ECT, and displaying this new ECT in the list of ECTs added to a school.

### Ticket

closes #478 

### Changes 

- As described above
- Destroy the session when a user begins the register ECT journey

### Video

https://github.com/user-attachments/assets/e50f5fb3-0788-44d4-9936-6a0eb2f70ae4

### CODE REVIEW

File: app/views/schools/register_ect/email_address.html.erb [hunk 1]
```diff
@@ -7,5 +7,5 @@
 
   <%= f.govuk_email_field(:email, label: { text: "Enter your ECT's email address.", class: 'govuk-!-margin-bottom-4' }) %>
 
-  <%= f.govuk_submit "Continue" %>
+  <%= f.govuk_submit "Continue", "data-testid": "submit" %>
```
Comment (peteryates) [inline app/views/schools/register_ect/email_address.html.erb:10]: What is `data-testid` doing here?

File: .github/workflows/update-ticket.yml
Comment (peteryates) [inline .github/workflows/update-ticket.yml:1]: I think this should be a separate PR. I'm not really a fan of including issue numbers in PR names as [GitHub supports this natively in a number of ways already](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue), but if others on the team would find it useful too we can add it.

File: app/views/schools/register_ect/email_address.html.erb (hunk 1 above)
Comment (tonyheadford) [reply app/views/schools/register_ect/email_address.html.erb:10]: Yes, not sure this actually helps much looking at the code, and as was mentioned by @peteryates if we do use the content in the selectors at least we know when something has changed.

File: .github/workflows/update-ticket.yml
Comment (tonyheadford) [reply .github/workflows/update-ticket.yml:1]: Agree, should at least be a separate PR, but I'm not sure of the value either when everything is in the same tool i.e. it is helpful when the issues/tickets are in another (Github aware) app but maybe not so much now everything is in Github?

File: app/views/schools/register_ect/email_address.html.erb (hunk 1 above)
Comment (TobyRet) [reply app/views/schools/register_ect/email_address.html.erb:10]: Sorry I may have misunderstood the discussion we had on this. I though Pete did not care either way and Tony mentioned that you were moving to data-testids in ECF1 (as an ongoing exercise). 

Happy to revert to text selectors if that is the consensus.

File: .github/workflows/update-ticket.yml
Comment (TobyRet) [reply .github/workflows/update-ticket.yml:1]: No problem - just an idea and came about from a discussion with Claire. I can delete. We can use the native support as Pete mentioned but this will still be a manual process in the sense it will rely on the author remembering to link the issue.

File: app/views/schools/register_ect/email_address.html.erb (hunk 1 above)
Comment (TobyRet) [reply app/views/schools/register_ect/email_address.html.erb]: testIds reverted

File: .github/workflows/update-ticket.yml
Comment (TobyRet) [reply .github/workflows/update-ticket.yml:1]: worflow changes reverted

File: app/views/schools/register_ect/email_address.html.erb (hunk 1 above)
Comment (tonyheadford) [reply app/views/schools/register_ect/email_address.html.erb]: Yes you're right, we did have a ticket in ECFv1 for it, but it'll never happen now.  I'm pretty ambivalent about it tbh, but I do like the idea of catching extra issues too. Thanks for reverting

**Review by tonyheadford — APPROVED** [review]
LGTM!
